demo/admin/src/dam/ImportFromUnsplash.tsx

```diff
@@ -0,0 +1,85 @@
+import { useApolloClient } from "@apollo/client";
+import { CancelButton, messages, SaveButton } from "@comet/admin";
+import { useDamAcceptedMimeTypes } from "@comet/cms-admin";
+import { useFileUpload } from "@comet/cms-admin/lib/dam/DataGrid/fileUpload/useFileUpload";
+import { clearDamItemCache } from "@comet/cms-admin/lib/dam/helpers/clearDamItemCache";
+import { Button, Dialog, DialogActions, DialogContent, DialogTitle } from "@mui/material";
+import { styled } from "@mui/material/styles";
+import * as React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { getRandomUnsplashImage, UnsplashImage } from "../dam/getRandomUnsplashImage";
+import UnsplashIcon from "./UnsplashIcon";
+
+export const ImportFromUnsplash: React.FC = () => {
+    const client = useApolloClient();
+    const { allAcceptedMimeTypes } = useDamAcceptedMimeTypes();
+    const [isOpen, setIsOpen] = React.useState(false);
+    const [unsplashImage, setUnsplashImage] = React.useState<UnsplashImage>();
+
+    const { uploadFiles } = useFileUpload({
+        acceptedMimetypes: allAcceptedMimeTypes,
+        onAfterUpload: () => {
+            client.reFetchObservableQueries();
+            clearDamItemCache(client.cache);
+        },
+    });
+
+    const handleOpenDialog = async () => {
+        const image = await getRandomUnsplashImage();
+        setUnsplashImage(image);
+        setIsOpen(true);
+    };
+
+    const handleCloseDialog = () => {
+        setIsOpen(false);
+    };
+
+    const handleSave = () => {
+        if (unsplashImage === undefined) return;
+        uploadFiles(
+            { acceptedFiles: [unsplashImage.file], fileRejections: [] },
+            {
+                importedFile: {
+                    sourceId: unsplashImage.url,
+                    sourceType: "unsplash",
+                },
```

Comment:
What happens when you use this button inside a folder? It should import the file to the correct folder

Reply:
@thomasdax98 can we use the same context provider for UploadFilesButton, which also needs the folder ID?